ext/standard/basic_functions.stub.php

```diff
@@ -177,34 +181,44 @@ function array_unique(array $arg, int $flags = SORT_STRING): array {}
 
 function array_intersect_key(array $arr1, array $arr2, array ...$arrays): array {}
```

Comment:
Is the `array` native type for the `$arrays` parameters correct? It doesn't look right for me.

Reply:
Looks right to me, why do you think it's wrong?

Reply:
My understanding is that you have to add the proper ZPP rule in order to be eligible for adding an argument type (so that passing incorrect types will fail during ZPP). Since the last parameter is variadic, a custom check is needed right after ZPP to ensure that only arrays are provided, so the arginfo/zpp checks in debug mode would fail. What do I miss?

Reply:
The requirement is that if the signature check fails, the function must throw. The exception does not necessarily have to be generated by zpp though, it just needs to be compatible with its semantics.

Reply:
Ah, got it now.